### ISSUE TITLE
Exercise 1 

### ISSUE THREAD

**Issue by miguel9554** [miguel9554/EAMTA2023_ML_track#1]
At commit f32a1bc3db37961466b61856b3a0f5241a3fbb87  we are at a test accuracy of ~65%, around 20% lower than the target 85%.
Some possibilities to improve

- Reduce layers kernel size: this mostly prevents overfitting
- Increase layers output channels: this allows for detection of more complex patterns
- Increas layers: this gives more capacity to the model to learn hierarchical structures in the data
- Increase training epochs: this gives more robustness and performance, but increasing it may also inccur in overfitting
- Use data augmentation: apply a random transformation to the input data, to improve feature detection

We observe following evolution of accuracies

![image](https://user-images.githubusercontent.com/31973644/223221130-05f0db1a-cb17-45df-93fc-d283eb4d451b.png)

We suspect overfitting, since training accuracy increases while test accuracy stays constant OR even decreases

**Comment by miguel9554** [comment]
Added a third layer @ 6671e24afc5865fd4e2e41d41b2b927bbab2c515 , overfitting improved, but accuracy is still low

![image](https://user-images.githubusercontent.com/31973644/223224730-1ed364ef-c075-44ed-89a5-46bcccb13342.png)

Actually, training accuracy curve is more increasing, but the final value did not changed, < 65%

**Comment by itdepedro** [comment]
Add an HorizontalFlipTransformation @ 9a41515130591f8bc13a471cd243e55477660f21 adding transforms.RandomHorizontalFlip() generates that training accuray and test accuracy grow together up to epoch 8. However the final values are worst than innital commit

![image](https://user-images.githubusercontent.com/27791699/223227153-e8147ae8-4499-4d59-b485-88053ed0cdf3.png)

```
==========================================================================================
       | Train Accuracy:  71.22% | Test Accuracy:  67.47%
   10  |----------------------------------------------------------------------------------
       | Train Loss:      1.6247 | Test Loss:      0.9569
==========================================================================================
TRAINING FINISHED
==========================================================================================
```

**Comment by miguel9554** [comment]
Using colorJitter data augmentation transformation (bd6585611ba59e25a872396901d617a09a496be5) accuracy improved but overfitting is still present

![image](https://user-images.githubusercontent.com/31973644/223425409-65a463c7-2add-48b2-93e9-3be0f42142c5.png)

**Comment by miguel9554** [comment]
Increasing layers to 3 and removing stride from a pool operator (815fdb6aff8fe1e194ec21ab4f5d51fdaaa320e6 ), we obtained better accuracy and less overfitting

Also, batch size was increased to 32

![image](https://user-images.githubusercontent.com/31973644/223510664-c6e9b651-f6cc-4d97-be9e-a64dbe735035.png)

Final test accuracy is at 76.53%

```
==========================================================================================
       | Train Accuracy:  82.66% | Test Accuracy:  76.53%
   20  |----------------------------------------------------------------------------------
       | Train Loss:      0.9842 | Test Loss:      0.6912
==========================================================================================
2023-03-07 14:47:22.570520 -> Epoch training time: 05:58
==========================================================================================
TRAINING FINISHED
==========================================================================================
2023-03-07 14:47:22.571520 -> Total training time: 114:34
2023-03-07 14:47:22.571520 -> Average epoch training time: 05:43
==========================================================================================
```

**Comment by miguel9554** [comment]
Increasing batch size, adding padding and doing pool with stride always (817f748bfa85fb86db780a82d7eb92d6cc417c6c), we get 77% test accuracy at a reduced parameter count of 350,590

![image](https://user-images.githubusercontent.com/31973644/223534645-ba312406-0137-4398-b447-7cfd7e991483.png)


```
==========================================================================================
       | Train Accuracy:  83.23% | Test Accuracy:  77.02%
   20  |----------------------------------------------------------------------------------
       | Train Loss:      0.9534 | Test Loss:      0.7106
==========================================================================================
2023-03-07 16:35:43.877762 -> Epoch training time: 03:04
==========================================================================================
TRAINING FINISHED
==========================================================================================
2023-03-07 16:35:43.877762 -> Total training time: 61:25
2023-03-07 16:35:43.878760 -> Average epoch training time: 03:04
==========================================================================================
```

It looks like more epochs could result in better performance

**Comment by miguel9554** [comment]
Increasing epochs to 40 (1b6c422f8ab178db76bb7474c377e56e9bba7b7d) did a marginal increase on accuracy but shows some overfitting

![image](https://user-images.githubusercontent.com/31973644/223722328-ee1fe3ca-0bcb-4771-936b-abe12b8fbb16.png)


```
==========================================================================================
       | Train Accuracy:  89.37% | Test Accuracy:  78.51%
   40  |----------------------------------------------------------------------------------
       | Train Loss:      0.5945 | Test Loss:      0.6899
==========================================================================================
2023-03-07 18:43:48.740718 -> Epoch training time: 03:00
==========================================================================================
TRAINING FINISHED
==========================================================================================
2023-03-07 18:43:48.740718 -> Total training time: 120:56
2023-03-07 18:43:48.740718 -> Average epoch training time: 03:01
==========================================================================================

Process finished with exit code 0
```
